### DISCUSSION TITLE
Table headers + Add button when using "DetailRowTemplate"

### DISCUSSION THREAD

**Discussion by wcrisafu** [Megabit/Blazorise#4363] [Q&A]
Hi,

I am using the "DetailRowTemplate" to show related records and I would like to show the headers of the "sub-grid". + If I make the DetailRowTemplate grid editable, I can see and use the Edit, Delete, Save buttons, but not the "Add" button. Do you have any example that would help?

Regards,

William

**Comment by David-Moreira** [comment]
Hello,
The New Button is usually located in the header since it isn't exactly row specific:
https://bootstrapdemo.blazorise.com/tests/datagrid/editing/commands

My guess is that for the `DetailRowTemplate` you're using another `Datagrid` inside it... You just have to treat it as you would any `Datagrid`...
If you want to move the `New` button to other places. you can call `datagridRef.New`.

https://bootstrapdemo.blazorise.com/tests/datagrid/editing/button-row

**Reply by wcrisafu** [reply]
Ho,

You are correct. I am using the "DetailRowTemplate" and a new DataGrid inside a main datagrid. I already found the way to add custom butons. Do you know of a way to force the sub-grid headers?

**Reply by David-Moreira** [reply]
Can you elaborate on the sub grid headers?
